pendulum/duration.py

```diff
@@ -414,6 +414,29 @@ def __divmod__(self, other):
 
         return NotImplemented
 
+    def isoformat(self) -> str:
+        """Represent this duration as a ISO-8601-compliant string."""
+        periods = [
+            ("Y", self.years),
+            ("M", self.months),
+            ("D", self.remaining_days),
+        ]
+        period = "P"
+        for sym, val in periods:
+            period += "{val}{sym}".format(val=val, sym=sym)
```

Comment:
if you only keep the components that are set, `.isoformat()` becomes a proper inverse transformation of `str → Duration`

```python
assert pendulum.parse("P1W").isoformat() == "P1W"
```

need same change below obviously